### PR TITLE
Additional properties

### LINKED ISSUE #232: Handle additionalProperties: true
It appears there is an `additional-properties.json` in the testserver which we haven't generated a client for. Let's do that, implement the tests, and check the nodejs runtime/generator for any special handling that might have been added there but didn't make it over here.

### PR DESCRIPTION
**Pull request by RikkiGibson**
Resolves #232

### CODE REVIEW

File: src/vanilla/ClientModelExtensions.cs [hunk 1]
```diff
@@ -896,6 +893,32 @@ void applyConstraints(TSObject obj)
                             }
                         });
                     }
+
+                    if (composite.AdditionalProperties != null)
+                    {
+                        typeObject.ObjectProperty("additionalProperties", additionalProperties =>
+                        {
+
```
Comment (daschult) [inline src/vanilla/ClientModelExtensions.cs]: Remove extra whitespace.

File: test/azuremetadata/generated/Lro/README.md [hunk 2]
```diff
@@ -1,80 +1,80 @@
...
-    </script>
-  </head>
-  <body>
-  </body>
-</html>
-```
-
-# Related projects
- - [Microsoft Azure SDK for Javascript](https://github.com/Azure/azure-sdk-for-js)
+# Microsoft Azure SDK for isomorphic javascript - AutoRestLongRunningOperationTestService
```
Comment (RikkiGibson) [inline test/azuremetadata/generated/Lro/README.md]: Wish all this line endings junk wasn't in here. But I don't believe meaningful changes happened.

File: src/vanilla/Model/CompositeTypeTS.cs [hunk 3]
```diff
@@ -67,6 +67,38 @@ public IReadOnlyList<CompositeTypeTS> ImmediatePolymorphicSubtypes
...
+                var type = AdditionalProperties.TSType(true);
+                if (type != "any")
+                {
+                    result += $"The value of an unknown property MUST be of type \"{type}\". Due to valid TS constraints " +
+                        $"we have modeled this as a union of `{type} | any`.";
+                }
+                else
+                {
+                    result += "The value of an unknown property can be of \"any\" type.";
+                }
```
Comment (daschult) [inline src/vanilla/Model/CompositeTypeTS.cs]: Is it possible for AdditionalProperties to have a description property specified? If so, then I'd add that description first.

Comment (RikkiGibson) [reply src/vanilla/Model/CompositeTypeTS.cs]: The AdditionalProperties property is created by the modeler. It has the generated description "Unmatched properties from the message are deserialized this collection".

It doesn't appear to be possible to add a description for the `"additionalProperties"` in Swagger, so I think I'm content with this approach.

**Pull request merged**